### PR TITLE
Columns: replace SHOULD with MUST + Condition

### PR DESCRIPTION
**Pull request by jpradocueva**
TF-2 agreed to prepare a PR to address all the current Conditional column requirements. 
The proposal is to change "SHOULD" for "MUST" + Condition.

### CODE REVIEW

File: specification/columns/commitmentdiscountcategory.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
...
 
-The CommitmentDiscountCategory column SHOULD be present in the billing data when the provider supports *commitment-based discounts*. This column MUST be of type String, MUST be null when [CommitmentDiscountId](#commitmentdiscountid) is null, and MUST NOT be null when CommitmentDiscountId is not null. The CommitmentDiscountCategory MUST be one of the allowed values.
+The CommitmentDiscountCategory column MUST be present in the billing data when the provider supports *commitment-based discounts*. This column MUST be of type String data and MUST contain one of the allowed enumerated values:
+* "Spend": Commitment-based discounts that require a predetermined amount of spend.
+* "Usage": Commitment-based discounts that require a predetermined amount of usage.
+
+Additionally:
+ IF [CommitmentDiscountId](#commitmentdiscountid) is null, THEN CommitmentDiscountCategory MUST also be null.
+ IF [CommitmentDiscountId](#commitmentdiscountid) is not null, THEN CommitmentDiscountCategory MUST NOT be null and adhere to the allowed enumerated values.
```
Comment (udam-f2) [inline specification/columns/commitmentdiscountcategory.md:11]: Group discussed minimizing the change to just the wording update
```suggestion
The CommitmentDiscountCategory column MUST be present in the billing data when the provider supports *commitment-based discounts*. This column MUST be of type String, MUST be null when [CommitmentDiscountId](#commitmentdiscountid) is null, and MUST NOT be null when CommitmentDiscountId is not null. The CommitmentDiscountCategory MUST be one of the allowed values.
```

File: specification/columns/skupriceid.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ A unique identifier that defines the unit price used to calculate the charge.
 | Constraint       | Value          |
 | :--------------- | :------------- |
 | Column type      | Dimension      |
-| Feature level    | Conditional    |
+| Feature level    | Mandatory      |
```
Comment (udam-f2) [inline specification/columns/skupriceid.md:24]: Looks like this got screwed up in a previous commit for ChargeCategory fixes. It should say Conditional and also have the conditional sentence in the normative paragraph.

**Comment by udam-f2** [comment]
https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/355/files brought in the SKUPriceId change. Looks like we need to also update the table in ChargeClass given that it still has the 'Column required' text.

File: specification/columns/skupriceid.md (hunk 2 above)
Comment (jpradocueva) [reply specification/columns/skupriceid.md:24]: @udam-f2, can you confirm whether the condition listed below is correct?

"The SkuPriceId column MUST be present in the billing data when the provider provides a SKU list ...."

**Review by jpradocueva — COMMENTED** [review]
@udam-f2 could you please check whether the condition is correct for the column, skupriceid? Thanks

**Comment by ijurica** [comment]
I believe this is the lost condition for the SkuPriceId :)

* The SkuPriceId column MUST be present in the billing data when the provider publishes a SKU price list. 

And the targeted Feature level

*   Feature level: `Conditional`

File: specification/columns/skupriceid.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ A unique identifier that defines the unit price used to calculate the charge.
 | Constraint       | Value          |
 | :--------------- | :------------- |
 | Column type      | Dimension      |
-| Feature level    | Conditional    |
+| Feature level    | Conditional      |
```
Comment (flanakin) [inline specification/columns/skupriceid.md:24]:
```suggestion
| Feature level    | Conditional    |
```

**Review by flanakin — APPROVED** [review]

**Comment by jpradocueva** [comment]
@mike-finopsorg, @udam-f2, must approve this PR before merging it.

**Review by udam-f2 — APPROVED** [review]

File: specification/columns/skupriceid.md (hunk 2 above)
Comment (udam-f2) [reply specification/columns/skupriceid.md]: Looks good

**Pull request merged**